animated-placeholder-demo/main.js

```diff
@@ -68,36 +67,72 @@ search.addWidget(
 
-let searchBar = document.getElementsByClassName("ais-SearchBox-input")[0]; //Getting the searchbar input
-let counter = 0;
-
-function getRandomDelayBetween(min, max) {
-  return Math.floor(Math.random() * (max - min + 1) + min);
-}
-
-//Printing the placeholder text in a 'typing' effect
-function printLetter(string, input) {
-  let placeholderCharsArray = string.split(""); //Splitting the string into character seperated array
-  let originalString = string; //Storing the full placeholder
-  let currentPlaceholder = input.getAttribute("placeholder"); //Getting the current placeholder value
-  let newPlaceholder = currentPlaceholder + placeholderCharsArray[counter]; //Appending the next letter to current placeholder
-
-  setTimeout(function() {
-    input.setAttribute("placeholder", newPlaceholder); //Modifying the placeholder text
-    counter++;
-    //Looping until the placeholder is fully printed
-    if (counter < placeholderCharsArray.length) {
-      printLetter(originalString, input);
-    }
-  }, getRandomDelayBetween(50, 90)); //Using a random speed to simulate 'human' typing
-}
+const setPlaceholder = (inputNode, placeholder) => {
+  inputNode.setAttribute("placeholder", placeholder);
+};
+
+const animateLetters = (
+  currentLetters,
+  remainingLetters,
+  inputNode,
+  onAnimationEnd
+) => {
+  if (!remainingLetters.length) {
+    return (
+      typeof onAnimationEnd === "function" &&
+      onAnimationEnd(currentLetters.join(""), inputNode)
+    );
+  }
+
+  currentLetters.push(remainingLetters.shift());
+
+  setTimeout(() => {
+    setPlaceholder(inputNode, currentLetters.join(""));
+    animateLetters(currentLetters, remainingLetters, inputNode, onAnimationEnd);
+  }, getRandomDelayBetween(50, 90));
+};
+
+const animatePlaceholder = (inputNode, placeholder, onAnimationEnd) => {
+  animateLetters([], placeholder.split(""), inputNode, onAnimationEnd);
+};
 
-//Func to init the animation
-function animatePlaceholder() {
-  searchBar.setAttribute("placeholder", ""); //Removing the initial placeholder
-  printLetter(placeholder, searchBar); //Starting the animation
-}
+// Code if we we want to restart animation after it ends
+const DELAY_AFTER_ANIMATION = 1000;
+const PLACEHOLDERS = [
+  "Search for green hoodie",
+  "Search for our latest items",
+  "Find your favorite movie"
+];
 
-animatePlaceholder();
+const onAnimationEnd = (placeholder, inputNode) => {
+  setTimeout(() => {
+    let newPlaceholder =
+      PLACEHOLDERS[Math.floor(Math.random() * PLACEHOLDERS.length)];
+
+    do {
+      newPlaceholder =
+        PLACEHOLDERS[Math.floor(Math.random() * PLACEHOLDERS.length)];
+    } while (placeholder === newPlaceholder);
+
+    animatePlaceholder(inputNode, newPlaceholder, onAnimationEnd);
+  }, DELAY_AFTER_ANIMATION);
+};
+
+// With multiple different placeholders
+window.addEventListener("load", () => {
+  // Single placeholder option
+  animatePlaceholder(searchBar, "This is an animated placeholder");
```

Comment:
Let me know and I can make a PR

Reply:
I think for simplicity, this is ok. It's simple to understand, you have a fn that accepts an input and text to animate. If we make it an array at first, it will make the whole thing more complex without an unnecessary benefit imo.

Reply:
ok got it! You're right. Let's just leave it as is then. 

This guide can be broken down into 2 section - (1) Animate 1 placeholder (2) Animate multiple placeholders. We'll just make the PLACEHOLDER constant an array if the reader decided to implement the 2nd solution.